datasetIngestor/syncDataToFileserver_unix.go

```diff
@@ -21,27 +23,54 @@ func SyncDataToFileserver(datasetId string, user map[string]string, RSYNCServer
 	// append trailing slash to sourceFolder to indicate that the *contents* of the folder should be copied
 	// no special handling for blanks in sourceFolder needed here
 	fullSourceFolderPath := sourceFolder + "/"
-	// check if filelisting given
-	// rsync can create only one level deep directory structure, here we need more, therefore mkdir -p
-	// This code is no longer needed, sine Edgar has a new rrsync wrapper which craetes the needed directory
-	// cmd := exec.Command("/usr/bin/ssh",RSYNCServer,"mkdir","-p",destFolder)
-	// // show rsync's output
-	// cmd.Stdout = os.Stdout
-	// cmd.Stderr = os.Stderr
-	//
-	// fmt.Printf("Running %v.\n", cmd.Args)
-	// cmd.Run()
+	
+	versionNumber, err := getRsyncVersion()
+	if err != nil {
+		log.Fatal("Error getting rsync version: ", err)
+	}
+	
+	rsyncCmd := buildRsyncCmd(versionNumber, absFileListing, fullSourceFolderPath, serverConnectString)
+		
+	// Show rsync's output	
+	rsyncCmd.Stderr = os.Stderr
+	log.Printf("Running: %v.\n", rsyncCmd.Args)
+	err = rsyncCmd.Run()
+	return err
+}
 
-	cmd := exec.Command("/usr/bin/rsync", "-e", "ssh -q", "-avxz", "--progress", "--msgs2stderr", fullSourceFolderPath, serverConnectString)
-	// // TODO: create folderstructure mkdir -p also for this case:
-	if absFileListing != "" {
-		cmd = exec.Command("/usr/bin/rsync", "-e", "ssh -q", "-avxzr", "--progress", "--msgs2stderr", "--files-from", absFileListing, fullSourceFolderPath, serverConnectString)
+// Get rsync version
+func getRsyncVersion() (string, error) {
+	cmd := exec.Command("/usr/bin/rsync", "--version")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	version := string(output)
+	
+	// Use a regular expression to find the version number.
+	// It will match the first occurrence of a string in the format "x.y.z" in the `version` string, where "x", "y", and "z" are one or more digits.
+	re := regexp.MustCompile(`\d+\.\d+\.\d+`)
+	versionNumber := re.FindString(version)
+	if versionNumber == "" {
+		return "", fmt.Errorf("could not find version number in rsync version string: %s", version)
 	}
-	// show rsync's output
-	// cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	
+	return versionNumber, nil
+}
 
-	log.Printf("Running %v.\n", cmd.Args)
-	err = cmd.Run()
-	return err
+// Check rsync version and adjust command accordingly
+func buildRsyncCmd(versionNumber, absFileListing, fullSourceFolderPath, serverConnectString string) *exec.Cmd {
+	rsyncFlags := []string{"-e", "ssh", "-avxz", "--progress"}
+	if absFileListing != "" {
+		rsyncFlags = append([]string{"-r", "--files-from", absFileListing}, rsyncFlags...)
+	}
+	if version.Compare(versionNumber, "3.2.3", ">=") {
+		rsyncFlags = append(rsyncFlags, "--stderr=error")
+		// Full command: /usr/bin/rsync -e ssh -avxz --progress -r --files-from <absFileListing> --stderr=error <fullSourceFolderPath> <serverConnectString>
+	} else {
+		rsyncFlags = append(rsyncFlags, "-q", "--msgs2stderr")
```

Comment:
My mac has v 3.1.3 but doesn't support `--msgs2stderr`.

Reply:
Maybe we should reduce the threshold from 3.2.3?

Reply:
I'm not sure what to do here.

Reply:
Let's leave it as an open issue